jobserver/authorization/forms.py

```diff
@@ -13,8 +13,8 @@ class RolesForm(forms.Form):
     classes on save.
     """
 
-    def __init__(self, available_roles, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, *, available_roles, **kwargs):
```

Comment:
What magic is this!?

Reply:
Keyword-only arguments! [PEP](https://www.python.org/dev/peps/pep-3102/)

It forces all args to be used as keyword arguments when called, great for making interfaces very explicit.

Reply:
I'm not sure I understand. I know that `*args` will be a `list` of as many arguments as are supplied and `**kwargs` will be a `dict` of as many keyword arguments as are supplied. I'm unsure about the single asterisk, though. Are we saying "Pass as many arguments as you like; only the last one matters"?

Reply:
The PEP explains the various details, but in essence the unnamed single asterisk means "end of positional arguments". So it's an error to supply any positional arguments at all to this method.

Reply:
Wowzers, I didn't know that. I'll read the PEP. Thanks both!